### ISSUE TITLE
Error from "Function should be wrap/unwrap" test

### ISSUE THREAD

**Issue by sestegra** [a14n/dart-js-wrapping#16]
**Environment**
- Dart SDK 1.24.2
- Packages
```
analyzer-0.29.11
args-0.13.7
async-1.13.3
barback-0.15.2+11
boolean_selector-1.0.2
browser-0.10.0+2
build-0.9.3
build_barback-0.3.0
build_runner-0.3.4+1
charcode-1.1.1
cli_util-0.0.1+2
code_transformers-0.5.1+3
collection-1.14.3
convert-2.0.1
crypto-2.0.2
csslib-0.14.0
dart_style-1.0.4
glob-1.1.3
html-0.13.2
http-0.11.3+14
http_multi_server-2.0.3
http_parser-3.1.1
isolate-1.0.0
js-0.6.1
js_wrapping-0.4.0
logging-0.11.3+1
matcher-0.12.1+1
meta-1.1.1
mime-0.9.3
node_preamble-1.4.0
package_config-1.0.2
package_resolver-1.0.2
path-1.4.2
plugin-0.2.0+1
pool-1.3.1
pub_semver-1.3.2
shelf-0.6.8
shelf_packages_handler-1.0.0
shelf_static-0.2.5
shelf_web_socket-0.2.1
source_gen-0.5.10+1
source_map_stack_trace-1.1.4
source_maps-0.10.4
source_span-1.4.0
stack_trace-1.8.0
stream_channel-1.6.1
stream_transform-0.0.9
string_scanner-1.0.2
term_glyph-1.0.0
test-0.12.24+2
typed_data-1.1.4
utf-0.9.0+3
watcher-0.9.7+3
web_socket_channel-1.0.5
when-0.2.0
which-0.1.3
yaml-2.1.12
```

**Scenario**
```
$ dart tool/build.dart
$ pub run test -p dartium
0:10 +45 -7: test/types_handling_test.dart: Function should be wrap/unwrap [E]
  type '(int) => dynamic' is not a subtype of type 'JsFunction' of 'f' where
    JsFunction is from dart:js

  types_handling_test.g.dart 67:27           __codec3.<fn>
  package:js_wrapping/util/codec.dart 46:56  _Converter.convert
  dart:convert                               Codec.decode
  types_handling_test.g.dart 117:41          A.simpleFunc
  types_handling_test.dart 97:14             main.<fn>

```

**Comment by a14n** [comment]
Is it fixed by #15 ?

**Comment by sestegra** [comment]
Unfortunately not
I opened this issue because of it.
